src/core/lib/security/credentials/credentials.h

```diff
@@ -100,6 +100,8 @@ void grpc_override_well_known_credentials_path_getter(
 struct grpc_channel_credentials
     : grpc_core::RefCounted<grpc_channel_credentials> {
  public:
+  // \a type is used to downcast grpc_channel_credentials objects.
```

Comment:
Let's be a little more explicit about the usage and requirements here:
- The *pointer value* is used to uniquely identify a creds implementation for down-casting purposes.
- Every creds impl should use a unique string instance, and it must use the same unique string instance for every creds instance of that creds impl.

Reply:
> Every creds impl should use a unique string instance, and it must use the same unique string instance for every creds instance of that creds impl.

I think it's fine for an impl to use a different string if it wants to for some reason.

Reply:
I can't think of a case where we'd actually want to do that.  Can you?

Reply:
No, but I don't see a reason to enforce an additional requirement either

Reply:
I'm not talking about enforcing a requirement, I'm talking about documenting the API contract.  The contract here is that the pointer address of the string will be used to deternine implementation equality.  That means that an implementation should always return the same pointer for all of its instances.